_source/_posts/2022-10-20-spring-vault.md

```diff
@@ -0,0 +1,642 @@
+
+```shell
+vault token create -policy=vault-demo-app-policy
+```
+
+```shell
+WARNING! The following warnings were returned from Vault:
+
+  * Endpoint ignored these unrecognized parameters: [display_name entity_alias
+  explicit_max_ttl num_uses period policies renewable ttl type]
+
+Key                  Value
+---                  -----
+token                hvs.CAESIKd9pYyc9xesiqmwvepdGiHlPEB53By...
+token_accessor       2pzRXmGhxChobbMtqeM5zZzM
+token_duration       768h
+token_renewable      true
+token_policies       ["default" "vault-demo-app-policy"]
+identity_policies    []
+policies             ["default" "vault-demo-app-policy"]
+```
+
+**NOTE**: I could not find documentation about the warning _Endpoint ignored these unrecognized parameters_. It seems `vault CLI` is sending default parameters not required by the target API in this case. The command equivalent API call can be displayed using the `-output-curl-string` flag after the subcommand, for example:
+
+```shell
+vault token create -output-curl-string -policy=vault-demo-app-policy
+```
+The output string will look like below:
+
+```shell
+curl -X POST -H "X-Vault-Token: $(vault print token)" -H "X-Vault-Request: true" -d '{"policies":["vault-demo-app-policy"],"ttl":"0s","explicit_max_ttl":"0s","period":"0s","display_name":"","num_uses":0,"renewable":true,"type":"service","entity_alias":""}' http://127.0.0.1:8200/v1/auth/token/create
+```
+
+You are now ready to update the config server. In the `vault-config-server` project, edit `src/main/resource/application.yml` to add Vault as the config backend:
+
+```yml
+server:
+  port: 8888
+
+spring:
+  profiles:
+    active: vault
+  cloud:
+    config:
+      server:
+        vault:
+          host: 127.0.0.1
+          port: 8200
+          kvVersion: 2
+logging:
+  level:
+    org.springframework.web.client: TRACE
+```
+
+Note that the logging level is set to TRACE for the web client, to see the interaction between the server and Vault. Restart the `vault-config-server`.
+
+```shell
+./mvnw spring-boot:run
+```
+
+You should see the logs below if the server was configured correctly:
+
+```
+2022-09-23 01:19:10.105  INFO 14072 --- [main] SpringBootConfigurationServerApplication:
+ Started SpringBootConfigurationServerApplication in 4.525 seconds (JVM running for 4.859)
+```
+
+You will not see the config server trying to connect to Vault in the logs. That will happen when a config client requests the properties. Start the `vault-demo-app`, passing the token just created in the environment variable `SPRING_CLOUD_CONFIG_TOKEN`. For example:
+
+```shell
+SPRING_CLOUD_CONFIG_TOKEN=hvs.CAESIKd9pYyc9xesiqmwvep... \
+./mvnw spring-boot:run
+```
+
+When the `vault-demo-app` starts, it will request the configuration to the config server, which in turn will make REST requests to Vault. In the config server logs, with enough logging level, you will be able to see those requests:
```

Comment:
```suggestion
When the `vault-demo-app` starts, it will request the configuration to the config server, which in turn will make REST requests to Vault. In the config server logs, with an increased logging level, you will be able to see those requests:
```